### PR TITLE
MP-4315 ✨ Webhooks

### PR DESCRIPTION
**Pull request by M4tini**
https://myparcelcombv.atlassian.net/browse/MP-4315
---
A lot has changed in this PR. I basically changed half the plugin to make it work with a separate `myparcelcom_data` table instead of the old columns added to several Magento core tables. Also a webhook endpoint is added and updating the data in this new table.

### CODE REVIEW

File: Controller/Adminhtml/Order/CreateMyParcelShipments.php [hunk 1]
```diff
@@ -61,25 +51,24 @@ private function massAction()
...
-        } catch (\Throwable $e) {
+            $this->messageManager->addSuccessMessage(sprintf(
+                __(MyParcelOrderCollection::SUCCESS_SHIPMENT_CREATED),
+                implode(', ', $this->orderCollection->getIncrementIds())
+            ));
+        } catch (Throwable $e) {
             $this->messageManager->addErrorMessage(__(MyParcelOrderCollection::ERROR_SHIPMENT_CREATE_FAIL) . ': ' . $e->getMessage());
         }
-
-        return $this;
```
Comment (zwaans) [inline Controller/Adminhtml/Order/CreateMyParcelShipments.php:83]: This function no longer needs to return anything?

File: Model/Webhook.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            if (isset($shipmentData['attributes']['tracking_url'])) {
+                $data->setTrackingUrl($shipmentData['attributes']['tracking_url']);
+            }
+
+            $data->setStatusCode($statusData['attributes']['code']);
+            $data->setStatusName($statusData['attributes']['name']);
+
+            $this->dataResource->save($data);
+
+            return 1;
```
Comment (zwaans) [inline Model/Webhook.php:86]: these return statements cant be boolean?

File: Model/Webhook.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * @param array $body
+     * @throws Exception
+     */
+    private function verifySecret($body)
+    {
+        $signature = $this->request->getHeader('X-MYPARCELCOM-SIGNATURE');
+
+        if (hash_hmac('sha256', json_encode($body), $this->config->getWebhookSecret()) !== $signature) {
```
Comment (zwaans) [inline Model/Webhook.php:100]: Maybe we can check this the other way around, so if something fails in those functions we always throw the error

**Review by zwaans — DISMISSED** [review]

File: Controller/Adminhtml/Order/CreateMyParcelShipments.php (hunk 1 above)
Comment (M4tini) [reply Controller/Adminhtml/Order/CreateMyParcelShipments.php:83]: I can actually put this stuff in the `execute()` function, I have refactored this in the next PR.

**Pull request merged**